server/tests/user.spec.js

```diff
@@ -0,0 +1,278 @@
+
+    it('Should test if new user has first name and last name', (done) => {
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .send(regularUserParam)
+        .expect(200)
+        .end((err, res) => {
+          regularUser = res.body.user;
+          regularToken = res.body.token;
+          expect(regularUser.firstname).to.exist;
+          expect(regularUser.lastname).to.exist;
+          done();
+        });
+    });
+
+    it('Should ensure that new user has a role', (done) => {
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .expect(201)
+        .end((err, res) => {
+          expect(regularUser).to.have.property('RoleId');
+          expect(regularUser.RoleId).to.equal(2);
+          done();
+        });
+    });
+
+    it('Should ensure user cannot be created if one of email or password is lacking.', (done) => {
+      testUserParam.email = null;
+      testUserParam.password = null;
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .send(testUserParam)
+        .expect(500)
+        .end((err, res) => {
+          expect(res.body.errors[0].message).to.equal('email cannot be null');
+          expect(res.body.errors[1].message).to.equal('password cannot be null');
+          done();
+        });
+    });
+  });
+
+  describe('Get: (/users/) - Get a user', () => {
+    it('should not return a user if id is invalid', (done) => {
+      request.get('/users/123')
+        .set({ 'x-access-token': adminToken })
+        .expect(404)
+        .end((err, res) => {
+          expect(res.body.message).to.be.equal('User Not Found');
+          done();
+        });
+    });
+
+    it('Should return all users', (done) => {
+      request.get('/users')
+        .set({ 'x-access-token': adminToken })
+        .expect(200)
+        .end((err, res) => {
+          expect(Array.isArray(res.body)).to.equal(true);
+          expect(res.body[0]).to.have.property('id');
+          expect(res.body[0]).to.have.property('firstname');
+          expect(res.body[0]).to.have.property('lastname');
+          expect(res.body[0]).to.have.property('username');
+          expect(res.body[0]).to.have.property('email');
+          expect(res.body[0]).to.have.property('RoleId');
+          done();
+        });
+    });
+
+    it('should return user with a correct id', (done) => {
+      request.get(`/users/${regularUser.id}`)
+        .set({ 'x-access-token': adminToken })
+        .end((error, response) => {
+          expect(response.status).to.equal(200);
+          expect(regularUser.email).to.equal(regularUserParam.email);
+          done();
+        });
+    });
+
+    it('should return the documents belonging to a user', (done) => {
+      request.get('/users/2/documents')
+      .set({ 'x-access-token': adminToken })
+      .end((error, response) => {
+        expect(response.status).to.equal(200);
+        expect(response.body.length).to.equal(0);
```

Comment:
Do you have another test spec for a situation that 1 or more documents are returned for a user?